Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -167,16 +168,18 @@ public class Klaviyo : NSObject {
     }
     
     /**
-     handlePush: Extracts tracking information from received push notification and sends the data to Klaviyo for push-tracking
-     analystics.
+     handlePush: Sends the push payload back to Klaviyo as an opened event
+        if the message originated from Klaviyo, as determined by presence of _k tracking parameters
      
      - Parameter userInfo: NSDictionary containing the push notification text & metadata
      */
     public func handlePush(userInfo: NSDictionary) {
-        if let metadata = userInfo["_k"] as? NSDictionary {
-            trackEvent(eventName: KLPersonOpenedPush, properties: metadata)
-        } else {
-            trackEvent(eventName: KLPersonOpenedPush, properties: userInfo as NSDictionary)
+        // Test it originated from klaviyo by checking presence of _k tracking parameters
+        if let body = userInfo["body"] as? NSDictionary, let _ = body["_k"] as? NSDictionary {
+            //But we still want to send the full payload, plus the push token
+            let properties = userInfo.mutableCopy() as! NSMutableDictionary
```

Comment:
❔ Are there any risk associated with force type casting `userInfo.mutableCopy()` to a `NSMutableDictionary` like a crash? Can we conditionally  type cast it?

Reply:
I don't think it should ever fail, but fair enough, it could be another condition in the if let above.